catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -135,6 +146,46 @@ public Metacard getMetacard(String id)
         return result.getMetacard();
     }
 
+    /**
+     * Returns all associated metacards given two relatable attributes.
+     *
+     * @param primaryFilterAttribute   The primary attribute to search against
+     * @param secondaryFilterAttribute The secondary attribute to search against
+     * @param primaryAttributeValue    A primary attribute value
+     * @param secondaryAttributeValue  A secondary attribute value
+     * @return A set of metacards, filtered by the given attributes and it's values
+     * @throws UnsupportedQueryException  Error
+     * @throws SourceUnavailableException Error
+     * @throws FederationException        Error
+     */
+    public List<Metacard> getAssociatedMetacardsByTwoAttributes(String primaryFilterAttribute,
```

Comment:
❓ Is this is directly related to the Note logic? Unless this has applications outside of notes, it should be moved into the NoteUtil class instead of the more generic EndpointUtil.

Reply:
While this is used by notes, there's nothing about this method that makes it specific to notes/annotations. It just retrieves metacards based on two caller-defined attributes and values.

Reply:
![image](https://user-images.githubusercontent.com/9013416/29850806-16fd4c58-8ce5-11e7-80b8-15595b9ba70a.png)

It's just right now its only for notes so i'd rather generify this if/when we need to instead of prematurely.

Reply:
I'll move it to NoteUtil